src/abi-metadata.ts

```diff
@@ -19,8 +19,10 @@ export interface AbiMetadata {
 }
 
 const metadataStore: WeakMap<{ new (): Contract }, Record<string, AbiMetadata>> = new WeakMap()
+const contractMap: Map<string, { new (): Contract }> = new Map()
```

Comment:
Does contractMap need to be a WeakMap as well? I know there are constraints around WeakMap keys.

Reply:
We are now using `WeakMap` with `Symbol` as key.